packages/edit-site/src/components/global-styles/ui.js

```diff
@@ -302,7 +302,7 @@ function GlobalStylesUI() {
 				<ScreenTypography />
 			</GlobalStylesNavigationScreen>
 
-			<GlobalStylesNavigationScreen path="/typography/font-sizes/">
+			<GlobalStylesNavigationScreen path="/typography/font-sizes">
```

Comment:
The trailing slash in the `path` was preventing `Navigator` from considering `/typography/font-sizes` as the parent screen to `/typography/font-sizes/:origin/:slug` when navigating back. Removing the trailing slack seems to fix the behavior